### ISSUE TITLE
Shutdown with mAP

### ISSUE THREAD

**Issue by kenrubiooo** [AlexeyAB/darknet#2125]
Hi @AlexeyAB,

I just noticed that when I use the -map for training, my laptop automatically shuts down even if my network width and height are just 416. I also tried using batch size is 8 but the same result happens. Now, I am training without -map. My laptop does not shut down anymore. What can be the source of this problem?

Thank you!

**Comment by AlexeyAB** [comment]
@kenrubiooo Hi,

* Do you mean that your laptop exactly shuts down? (is no just errors: seg-fault, out of memory...)

* Try to use the latest version from this repo.

* If it doesn't help, try to use this commit: https://github.com/AlexeyAB/darknet/tree/95773cfb423266b9ac6aeea54e862db5817b5447  Download this Darknet version: https://github.com/AlexeyAB/darknet/archive/95773cfb423266b9ac6aeea54e862db5817b5447.zip

* What GPU do you use?

* How many CPU RAM do you have?

* What parameters do you use in the Makefile?

* What versions of CUDA, cuDNN, OpenCV do you use?

* What command do you use for training?

**Comment by kenrubiooo** [comment]
Hi @AlexeyAB 

Sorry for the late reply. I updated my local repository. I ran the code without the -map. Now, it shuts down automatically, not just seg-fault or out-of-memory.

I am using a GTX1050 Ti. I also have 12 CPU RAM.

Makefile Parameters:

GPU=1
CUDNN=1
CUDNN_HALF=0
OPENCV=1
AVX=0
OPENMP=0
LIBSO=0
DEBUG=0

ARCH= -gencode arch=compute_30,code=sm_30 \
      -gencode arch=compute_35,code=sm_35 \
      -gencode arch=compute_50,code=[sm_50,compute_50] \
      -gencode arch=compute_52,code=[sm_52,compute_52] \
	  -gencode arch=compute_61,code=[sm_61,compute_61]

OS := $(shell uname)


CUDA version - V9.0.176
OpenCV version - 3.4.2

Training command:
I speedup the training by using:
./darknet partial cfg/yolo-seven.cfg yolov3.weights yolov3.conv.81 81

Then I start training with 
./darknet detector train /home/ubuntu/alt_darknet/build/darknet/x64/data/seven.data cfg/yolo-seven.cfg yolov3.conv.81

I am now at checkpoint 500.weights. After 150 iterations, it shuts down automatically. What could be the problem? Thank you!

**Comment by AlexeyAB** [comment]
@kenrubiooo 

May be this is an overheat issue, like there: https://github.com/AlexeyAB/darknet/issues/407#issuecomment-368872949
Try to compile with `GPU=1 CUDNN=0`.

**Comment by kenrubiooo** [comment]
My friends also told me that it might due to CPU overheating. Turning CUDNN to 0 will turn off the GPU training right? That will make the training very slow.

**Comment by AlexeyAB** [comment]
>  Turning CUDNN to 0 will turn off the GPU training right? 

No. It will turn off highly optimized cuDNN functions. But it will use less optimized CUDA-functions that are implemented in the Darknet.

**Comment by kenrubiooo** [comment]
Ok. I will try this tomorrow morning. I will update you. What do you think would be the best Makefile parameters in my condition?

Thank you!

**Comment by AlexeyAB** [comment]
@kenrubiooo Hi,

1. Also try to set here `args.threads = 1;` instead of
 https://github.com/AlexeyAB/darknet/blob/527578744b46666fb5cd42393bf9e1fa9af126ee/src/detector.c#L154

----

2. If it doesn't help, then try to set `#ifdef DONT_USE` instead of
https://github.com/AlexeyAB/darknet/blob/527578744b46666fb5cd42393bf9e1fa9af126ee/src/data.c#L731

and recompile.

**Comment by kenrubiooo** [comment]
Hi @AlexeyAB!

Here is an update. It still shuts down. I think it has other causes. The avg loss becomes `inf` then `nan` after some iterations, then shuts down.

I am training YOLOv3 with only 7 classes, each with 2000+ images. Each image only contains a single object. What I am confused with is that the count in these lines are most of the time not 1. I think it shoud be 1 only since there is only one object in every image

`Region 82 Avg IOU: 0.761891, Class: 0.604295, Obj: 0.932632, No Obj: 0.000719, .5R: 1.000000, .75R: 1.000000,  **count**: 1`
